nvfuser/contrib/nn/normalization.py

```diff
@@ -0,0 +1,741 @@
+import enum
+from typing import Any, Dict, List, Optional, Tuple
+
+import torch
+
+import nvfuser
+
+from nvfuser.pytorch_utils import torch_dtype_to_nvfuser_dtype
+
+
+__all__ = [
+    "InstanceNorm1dNVFuser",
+    "InstanceNorm2dNVFuser",
+    "InstanceNorm3dNVFuser",
+]
+
+
+NamedAxis = enum.Enum("NamedAxis", ["BATCH", "CHANNEL"])
+
+
+def partially_contig_tensor(
+    fd: "nvfuser.FusionDefinition",
+    x: torch.Tensor,
+) -> "nvfuser.Tensor":
+    return fd.define_tensor(
+        symbolic_sizes=[-1] * x.ndim,
+        contiguity=nvfuser.compute_contiguity(x.size(), x.stride()),
+        dtype=torch_dtype_to_nvfuser_dtype(x.dtype),
+    )
```

Comment:
Paging @kevinstephano who may want to weigh in on this.

There is a question of whether we should switch from `partially_contig_tensor` to `fd.from_pytorch`. The idea of `fd.from_pytorch` is to inspect the input sizes and strides and set the `IterType` and contiguity of the tensor based on those. When it sees a size of 1, it assumes that dimension should be a broadcast dimension and sets its symbolic size to `1` and contiguity to `None`. If it sees a size other than 1 it will set `symbolic_sizes=-1` and compute the contiguity as normal.

What I was trying to do with `partially_contig_tensor` is to skip setting anything as broadcast (never setting contiguity to `None`), since I knew that in this context, none of those dimensions will be used in a broadcast. The reason to do this is to enable re-using more fusions. For example, if someone runs with batch size 1 and then batch size N, they would not expect to face a recompilation, but they will when using `fd.from_pytorch`. Originally, I had my own `compute_contiguity` here which did not have a special case for broadcast, but that got converted to use the one from nvfuser which I think was later changed to support broadcast.

Probably a clean way to handle this is to have a `broadcast: Union[bool, List[bool]] = True` argument for `fd.from_pytorch`. If it's bool, we would infer whether each dimension is broadcast based on its size. If `False`, we would assume no dimensions are broadcast, and if it's a list, we'd use the values given but check that the (unexpanded) size is one for any `True` entries.